Source/FakeItEasy/Core/FakeWrapperConfigurator.cs

```diff
@@ -47,6 +47,13 @@ public IFakeOptionsBuilder<T> WithAdditionalAttributes(IEnumerable<CustomAttribu
 
         public IFakeOptionsBuilder<T> Implements(Type interfaceType)
         {
+            Guard.AgainstNull(interfaceType, "interfaceType");
+
+            if (!interfaceType.IsInterface)
+            {
+                throw new ArgumentException("The specified type is not an interface");
+            }
+

```

Comment:
Done - I've consolidated both checks to just ```FakeOptions.AddInterfaceToImplement``` and I've added the ```Type.Name``` to the ArgumentException.

Do you also want me to add a test for the new ```Guard.AgainstNull```?  Essentially it would be:
```
// Arrange, Act, Assert
NullGuardedConstraint.Assert(() => this.creator.CreateFake<IFoo>(x => x.Implements(null)));
NullGuardedConstraint.Assert(() => this.creator.CreateFake<IFoo>(x => x.Wrapping(A.Fake<IFoo>()).Implements(null)));
```